langchain4j-coherence/src/main/java/dev/langchain4j/store/embedding/coherence/CoherenceEmbeddingStore.java

```diff
@@ -0,0 +1,526 @@
+         */
+        private String sessionName;
+
+        /**
+         * The {@link Session} to use to obtain the {@link NamedMap}.
+         */
+        private Session session;
+
+        /**
+         * The {@link VectorIndexExtractor} to use to create a vector index used to query the {@link NamedMap}.
+         */
+        private VectorIndexExtractor<DocumentChunk, Vector<?>> extractor;
+
+        /**
+         * A flag that when {@code true} forces normalization of embeddings on adding and searching
+         */
+        private boolean forceNormalize = false;
+
+        /**
+         * Create a {@link CoherenceEmbeddingStore.Builder}.
+         */
+        protected Builder() {
+        }
+
+        /**
+         * Set the name of the {@link NamedMap} that will hold the
+         * {@link DocumentChunk document chunks}.
+         *
+         * @param name  the name of the {@link NamedMap} that will hold
+         *              the {@link DocumentChunk document chunks}
+         *
+         * @return this builder for fluent method calls
+         */
+        public Builder name(String name) {
+            this.name = name == null || name.isEmpty() ? DEFAULT_MAP_NAME : name;
+            return this;
+        }
+
+        /**
+         * Set the name of the {@link Session} to use to obtain the
+         * document chunk {@link NamedMap}.
+         *
+         * @param sessionName  the session name
+         *
+         * @return this builder for fluent method calls
+         */
+        public Builder session(String sessionName) {
+            this.sessionName = sessionName;
+            this.session = null;
+            return this;
+        }
+
+        /**
+         * Set the {@link Session} to use to obtain the
+         * document chunk {@link NamedMap}.
+         *
+         * @param session  the {@link Session} to use
+         *
+         * @return this builder for fluent method calls
+         */
+        public Builder session(Session session) {
+            this.session = session;
+            this.sessionName = null;
+            return this;
+        }
+
+        /**
+         * Set the vector index to add to the underlying {@link NamedMap}.
+         *
+         * @param extractor  the {@link VectorIndexExtractor} to use to create the index
+         *
+         * @return this builder for fluent method calls
+         */
+        public Builder index(VectorIndexExtractor<DocumentChunk, Vector<?>> extractor) {
+            this.extractor = extractor;
+            return this;
+        }
+
+        /**
+         * Set whether to force normalization of vectors on adding and searching.
+         *
+         * @param f  {@code true} if the {@link CoherenceEmbeddingStore} should call force
+         *           {@link Embedding#normalize()} on embeddings when adding or searching
+         *
+         * @return this builder for fluent method calls
+         */
+        public Builder forceNormalize(boolean f) {
```

Comment:
```suggestion
        public Builder normalizeEmbeddings(boolean f) {
```

nit: can we rename it? There is already `VearchEmbeddingStore` that uses `normalizeEmbeddings` name